solidity/app/contracts/Router.sol

```diff
@@ -130,22 +132,50 @@ abstract contract Router is AbacusConnectionClient, IMessageRecipient {
     }
 
     /**
-     * @notice Dispatches a message to an enrolled router via the local router's
-     * Outbox and pays for message processing on the destination chain.
-     * @dev Reverts if there is no enrolled router for _destination.
+     * @notice Pay for message processing on the destination
+     * @param _leafIndex The leaf index of the message to pay processing for
+     * @param _gasPayment The amount of native tokens to pay the Interchain Gas
+     * Paymaster to process the dispatched message.
+     */
+    function _payForGas(uint256 _leafIndex, uint256 _gasPayment) internal {
+        _interchainGasPaymaster().payGasFor{value: _gasPayment}(_leafIndex);
+    }
+
+    /**
+     * @notice Calls #checkpoint on the Outbox
+     */
+    function _checkpointOnOutbox() internal {
+        _outbox().checkpoint();
+    }
+
+    /**
+     * @notice A convenience function which allows the caller to 1) send a message
+     * to the enrolled router on _destination as specified on the ACM, 2) pay
+     * for message processing on the destination chain and 3) checkpoint the
+     * message on the Outbox
      * @param _destination The domain of the chain to which to send the message.
      * @param _msg The message to dispatch.
      * @param _gasPayment The amount of native tokens to pay the Interchain Gas
      * Paymaster to process the dispatched message.
+     * @param _shouldCheckpoint Whether checkpoint should be called on the Outbox
      */
-    function _dispatchToRemoteRouterWithGas(
+    function _comboDispatch(
         uint32 _destination,
         bytes memory _msg,
-        uint256 _gasPayment
+        uint256 _gasPayment,
+        bool _shouldCheckpoint
     ) internal {
-        uint256 leafIndex = _dispatchToRemoteRouter(_destination, _msg);
+        IAbacusConnectionManager _abacusConnectionManager = abacusConnectionManager;
```

Comment:
Why can't you just do:
```
IOutbox _outboxVar = _outbox();
```
?

Reply:
It's to save gas a la https://github.com/abacus-network/abacus-monorepo/pull/359#discussion_r858595112